src/liblsquic/lsquic_packet_common.h

```diff
@@ -50,7 +50,40 @@ enum quic_ft_bit {
     QUIC_FTBIT_PING             = 1 << QUIC_FRAME_PING,
 };
 
-extern const size_t lsquic_frame_types_str_sz;
+static const char * const frame_type_2_str[N_QUIC_FRAMES] = {
+    [QUIC_FRAME_INVALID]           =  "QUIC_FRAME_INVALID",
+    [QUIC_FRAME_STREAM]            =  "QUIC_FRAME_STREAM",
+    [QUIC_FRAME_ACK]               =  "QUIC_FRAME_ACK",
+    [QUIC_FRAME_PADDING]           =  "QUIC_FRAME_PADDING",
+    [QUIC_FRAME_RST_STREAM]        =  "QUIC_FRAME_RST_STREAM",
+    [QUIC_FRAME_CONNECTION_CLOSE]  =  "QUIC_FRAME_CONNECTION_CLOSE",
+    [QUIC_FRAME_GOAWAY]            =  "QUIC_FRAME_GOAWAY",
+    [QUIC_FRAME_WINDOW_UPDATE]     =  "QUIC_FRAME_WINDOW_UPDATE",
+    [QUIC_FRAME_BLOCKED]           =  "QUIC_FRAME_BLOCKED",
+    [QUIC_FRAME_STOP_WAITING]      =  "QUIC_FRAME_STOP_WAITING",
+    [QUIC_FRAME_PING]              =  "QUIC_FRAME_PING",
+};
+
+
+#define SLEN(x) (sizeof(#x) - sizeof("QUIC_FRAME_"))
```

Comment:
If we decide indeed to move this out to the header file, we should rename the macro to something more specific.